public/app/plugins/datasource/loki/responseUtils.ts

```diff
@@ -203,6 +203,10 @@ function combineFrames(dest: DataFrame, source: DataFrame) {
   const totalFields = dest.fields.length;
   for (let i = 0; i < totalFields; i++) {
     dest.fields[i].values = [].concat.apply(source.fields[i].values, dest.fields[i].values);
+    if (source.fields[i].nanos) {
+      const nanos: number[] = dest.fields[i].nanos?.slice() || [];
+      dest.fields[i].nanos = source.fields[i].nanos?.concat(nanos);
```

Comment:
By the way, I could not merge the fields in the same way we merge `values` because TS would refuse to understand that `nanos` was not undefined.

![ts](https://github.com/grafana/grafana/assets/1069378/a8beb30d-a8fa-4f7d-bc55-63950d28bb27)

Reply:
Would be really cool to have a `concatFrame` method somewhere in grafana/data at some point.

Reply:
Definitely. What do you think we can do to make that happen?